include/veriblock/blockchain/block_index.hpp

```diff
@@ -19,6 +19,25 @@
 
 namespace altintegration {
 
+enum BlockStatus : uint8_t {
+  //! default state for validity - validity state is unknown
+  BLOCK_VALID_UNKNOWN = 0,
+  //! acceptBlock succeded. All parents are at least at this state.
+  BLOCK_VALID_TREE = 1,
+  //! addPayloads succeded. All parents are at least BLOCK_VALID_TREE
+  BLOCK_VALID_PAYLOADS = 2,
+  //! all validity flags
+  BLOCK_VALID_MASK = BLOCK_VALID_TREE | BLOCK_VALID_PAYLOADS,
+  //! block is statelessly valid, but we marked it as failed
+  BLOCK_FAILED_BLOCK = 4,
+  //! block failed POP validation
+  BLOCK_FAILED_POP = 8,
+  //! block is state{lessly,fully} valid, but some of previous blocks is invalid
+  BLOCK_FAILED_CHILD = 16,
+  //! all invalidity flags
+  BLOCK_FAILED_MASK = BLOCK_FAILED_CHILD | BLOCK_FAILED_POP | BLOCK_FAILED_BLOCK
+};
+
```

Comment:
It seems it does not matter if it is BLOCK_FAILED_CHILD or BLOCK_FAILED_BLOCK or any other reason for a failure. Therefore in the long run these options will be messed up and used arbitrarily. I recommend to use a single FAILED flag.

Reply:
In current code we distinguish blocks that we marked as invalid manually (BLOCK_FAILED_BLOCK), or blocks that for descend from such block (BLOCK_FAILED_CHILD).
This will be used in future code, in command processing...